pkg/engine/engine.go

```diff
@@ -186,16 +181,21 @@ func (e *Engine) detectorWorker(ctx context.Context) {
 					if !foundKeyword {
 						continue
 					}
-					ctx, cancel := context.WithTimeout(ctx, time.Second*10)
-					defer cancel()
-					results, err := detector.FromData(ctx, verify, decoded.Data)
+
+					results, err := func() ([]detectors.Result, error) {
+						ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+						defer cancel()
+						defer common.Recover(ctx)
```

Comment:
Do you know if the order matters?

Reply:
Yea i think we want recover, then cancel. So with defer it's LIFO so I think this works.

Reply:
hm... or maybe we want that flipped.

Reply:
OKay yep cancel then recover looks correct.